### DISCUSSION TITLE
Should ecode ship with predefined build settings?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#158] [General]
At least for common programming languages like C++? The users will always be able to customize it to suit their own needs. Geany does ship with predefined rules for build and run. It makes everything very easy for everyone. I'm getting lost when trying to add new build settings to ecode. There is completely no examples nor documents to learn from.

**Comment by SpartanJ** [comment]
The problem is that ecode is completely generic regarding the language. Geany provides very simple rules to build a c or cpp file but it's not there to generate whole projects. For sure ecode can have some documentation that it's completely lacking for the build system. What it can be done in ecode is to provide some templates to build projects but I can't think an obvious solution.
